### DISCUSSION TITLE
Clarification on NeckAreaPostprocessor

### DISCUSSION THREAD

**Discussion by StoGitzo** [idaholab/moose#28119] [Q&A Modules: Phase field]
Hello MOOSE users!

I ran two-particle sintering simulations in a 2D domain and I tried to extend it to a 3D domain. For computational power issues, I considered 2 x 1/32 of particle without modifying any geometric or material-related parameter. 
The simulation was successful and, on a graphic level in Paraview, the 2D and 3D solutions coincide. However, when I plotted geometric and energy-related graphs, these are very different from the corresponding case in 2D.
I tried to do some debugging and among the Postprocessors I used, I cannot figure out how NeckAreaPostprocessor operates (and so ElementIntegralVariablePostprocessor), because I don't have much experience.

Thanks!
Simone

**Comment by GiudGiud** [comment]
Hello

I dont see this object in moose. Are you using the phase field module or are you using an application? if so which one?

Guillaume

**Reply by StoGitzo** [reply]
It comes from an old version of CrowApp by @SudiptaBiswas. The header and the .C are the following

```
#ifndef NECKAREAPOSTPROCESSOR_H
#define NECKAREAPOSTPROCESSOR_H

#include "ElementIntegralVariablePostprocessor.h"

//Forward Declarations
class NeckAreaPostprocessor;

/**
 * This postprocessor computes a volume integral of the specified variable.
 *
 * Note that specializations of this integral are possible by deriving from this
 * class and overriding computeQpIntegral().
 */
class NeckAreaPostprocessor :
  public ElementIntegralPostprocessor
{
public:
  static InputParameters validParams();
  NeckAreaPostprocessor(const InputParameters & parameters);

protected:
  virtual Real computeQpIntegral() override;

  const unsigned int _op_num;
  std::vector<const VariableValue *> _vals;
};

#endif //NECKAREAPOSTPROCESSOR_H
```

```
#include "NeckAreaPostprocessor.h"

registerMooseObject("ParrotApp", NeckAreaPostprocessor);

InputParameters
NeckAreaPostprocessor::validParams()
{
  InputParameters params = ElementIntegralVariablePostprocessor::validParams();
  params.addRequiredCoupledVarWithAutoBuild("v", "var_name_base", "op_num",
                                            "Array of coupled variables");
  return params;
}

NeckAreaPostprocessor::NeckAreaPostprocessor(const InputParameters &parameters)
    : ElementIntegralPostprocessor(parameters), _op_num(coupledComponents("v")),
      _vals(_op_num) {
  for (unsigned int i = 0; i < _op_num; ++i)
    _vals[i] = &coupledValue("v", i);
}

Real NeckAreaPostprocessor::computeQpIntegral() {
  Real gb = 0.0;
  for (unsigned int i = 0; i < _op_num; ++i)
    for (unsigned int j = 0; j < _op_num; ++j)
      if (i != j)
        gb = (*_vals[i])[_qp] * (*_vals[j])[_qp];

  // if (gb > 0.5 && _u[_qp] < 0.95)
  //   gb = 1.0;
  return gb;
}
```

**Reply by GiudGiud** [reply]
Hello

On every Qp it computes this gb quantity. 

It looks weird to me, the gb number gets overwritten multiple times in the for loop and only the last value is returned. 

@SudiptaBiswas was this intended?
